diff -rupN med-3.0.7_SRC/configure.ac med-3.0.7_SRC-new/configure.ac
--- med-3.0.7_SRC/configure.ac	2013-11-28 12:27:09.000000000 +0100
+++ med-3.0.7_SRC-new/configure.ac	2014-05-23 00:09:08.803332678 +0200
@@ -36,7 +36,7 @@ AC_CONFIG_MACRO_DIR([config])
 
 
 ## Initialize Automake
-AM_INIT_AUTOMAKE([dist-bzip2 tar-ustar serial-tests])
+AM_INIT_AUTOMAKE([dist-bzip2 tar-ustar])
 
 ##
 AC_CONFIG_HEADER(include/med_config.h)
