diff -rupN med-4.0.0/tests/unittests/c/MEDinterp/Test_MEDfieldInterpWr.shn med-4.0.0-new/tests/unittests/c/MEDinterp/Test_MEDfieldInterpWr.shn
--- med-4.0.0/tests/unittests/c/MEDinterp/Test_MEDfieldInterpWr.shn	2017-07-10 16:24:11.000000000 +0200
+++ med-4.0.0-new/tests/unittests/c/MEDinterp/Test_MEDfieldInterpWr.shn	2018-12-25 12:03:35.385823247 +0100
@@ -7,13 +7,13 @@ ${H5DUMP} ${BUILDDIR}/${file}.med > ${BU
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/${file}.dump > ${BUILDDIR}/${file}.dump.tmp && \
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/dumps.ref/${file}.dump > ${BUILDDIR}/${file}.dump.ref && \
 diff ${BUILDDIR}/${file}.dump.ref ${BUILDDIR}/${file}.dump.tmp && \
diff -rupN med-4.0.0/tests/unittests/c/MEDinterp/Test_MEDinterpBaseFunctionWr.shn med-4.0.0-new/tests/unittests/c/MEDinterp/Test_MEDinterpBaseFunctionWr.shn
--- med-4.0.0/tests/unittests/c/MEDinterp/Test_MEDinterpBaseFunctionWr.shn	2017-07-10 16:24:11.000000000 +0200
+++ med-4.0.0-new/tests/unittests/c/MEDinterp/Test_MEDinterpBaseFunctionWr.shn	2018-12-25 12:03:35.386823244 +0100
@@ -7,13 +7,13 @@ ${H5DUMP} ${BUILDDIR}/${file}.med > ${BU
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/${file}.dump > ${BUILDDIR}/${file}.dump.tmp && \
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/dumps.ref/${file}.dump > ${BUILDDIR}/${file}.dump.ref && \
 diff ${BUILDDIR}/${file}.dump.ref ${BUILDDIR}/${file}.dump.tmp && \
diff -rupN med-4.0.0/tests/unittests/c/MEDinterp/Test_MEDinterpCr.shn med-4.0.0-new/tests/unittests/c/MEDinterp/Test_MEDinterpCr.shn
--- med-4.0.0/tests/unittests/c/MEDinterp/Test_MEDinterpCr.shn	2017-07-10 16:24:11.000000000 +0200
+++ med-4.0.0-new/tests/unittests/c/MEDinterp/Test_MEDinterpCr.shn	2018-12-25 12:03:35.386823244 +0100
@@ -7,13 +7,13 @@ ${H5DUMP} ${BUILDDIR}/${file}.med > ${BU
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/${file}.dump > ${BUILDDIR}/${file}.dump.tmp && \
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/dumps.ref/${file}.dump > ${BUILDDIR}/${file}.dump.ref && \
 diff ${BUILDDIR}/${file}.dump.ref ${BUILDDIR}/${file}.dump.tmp && \
diff -rupN med-4.0.0/tests/unittests/c/Test_MEDstructElementCr.shn med-4.0.0-new/tests/unittests/c/Test_MEDstructElementCr.shn
--- med-4.0.0/tests/unittests/c/Test_MEDstructElementCr.shn	2017-07-10 16:24:11.000000000 +0200
+++ med-4.0.0-new/tests/unittests/c/Test_MEDstructElementCr.shn	2018-12-25 12:03:35.386823244 +0100
@@ -7,13 +7,13 @@ ${H5DUMP} ${BUILDDIR}/${file}.med > ${BU
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/${file}.dump > ${BUILDDIR}/${file}.dump.tmp && \
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/dumps.ref/${file}.dump > ${BUILDDIR}/${file}.dump.ref && \
 diff ${BUILDDIR}/${file}.dump.ref ${BUILDDIR}/${file}.dump.tmp && \
diff -rupN med-4.0.0/tests/unittests/c/Test_MEDsupportMesh.shn med-4.0.0-new/tests/unittests/c/Test_MEDsupportMesh.shn
--- med-4.0.0/tests/unittests/c/Test_MEDsupportMesh.shn	2017-07-10 16:24:11.000000000 +0200
+++ med-4.0.0-new/tests/unittests/c/Test_MEDsupportMesh.shn	2018-12-25 12:03:35.386823244 +0100
@@ -7,13 +7,13 @@ ${H5DUMP} ${BUILDDIR}/${file}.med > ${BU
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/${file}.dump > ${BUILDDIR}/${file}.dump.tmp && \
 sed -e 's/H5T_STD_I32BE//g' -e 's/H5T_STD_I32LE//g'	 \
     -e 's/H5T_STD_I64BE//g' -e 's/H5T_STD_I64LE//g'  \
     -e 's/H5T_IEEE_F64BE//g' -e 's/H5T_IEEE_F64LE//g'  \
-    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
+    -e 's/H5T_STD_I8LE//g' -e 's/H5T_STD_U8LE//g' -e 's/H5T_STD_I8BE//g' -e 's/H5T_STD_U8BE//g'  \
     -e 's/HDF5.*{//g' \
 ${BUILDDIR}/dumps.ref/${file}.dump > ${BUILDDIR}/${file}.dump.ref && \
 diff ${BUILDDIR}/${file}.dump.ref ${BUILDDIR}/${file}.dump.tmp && \
